### PR TITLE
[spaceship, deliver, snapshot] add support for new device  screenshot sizes (iPad Pro 11-inch and iPad Pro 12.9-inch 3rd Gen.) 

### PR DESCRIPTION
**Pull request by max-ott**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
This pull request adds support for the new iPad Pro 11-inch and iPad Pro 12.9-inch (3rd Gen.) as well as support for other "new" screenshots added by Apple in the past.

#### Snapshot
- Added support for 11 inch iPad

#### Deliver
- Added support for 11 inch iPad
- Added support for 12.9 inch iPad (3rd generation)

#### Spaceship
- Added picture type to `du_client`
- Added messages picture type to `du_client`
- Added resolutions to `du_client`
- Added device names to `device_type`
- Added video preview images to `tunes_client`

Also tried to "sort" devices where possible from lowest resolution to highest resolution in each of the different device families (iPhone / iPhone X / iPad / iPad Pro 2018) similar to how they get displayed in App Store Connect > Media Manager

### CODE REVIEW

**Comment by janpio** [comment]
```
Offenses:

spaceship/lib/spaceship/du/du_client.rb:128:60: C: Layout/TrailingWhitespace: Trailing whitespace detected.
        iPadPro11:    "MZPFT.SortedJ317MessagesScreenShot", 
                                                           ^
spaceship/lib/spaceship/du/du_client.rb:129:60: C: Layout/TrailingWhitespace: Trailing whitespace detected.
        iPadPro129:   "MZPFT.SortedJ320MessagesScreenShot", 
                                                           ^

1041 files inspected, 2 offenses detected
```

File: spaceship/lib/spaceship/du/du_client.rb [hunk 1]
```diff
@@ -121,13 +123,15 @@ def messages_picture_type_map
       # rubocop:enable Layout/ExtraSpacing
       {
         ipad:         "MZPFT.SortedTabletMessagesScreenShot",
-        ipadPro:      "MZPFT.SortedJ99MessagesScreenShot",
         ipad105:      "MZPFT.SortedJ207MessagesScreenShot",
+        ipadPro:      "MZPFT.SortedJ99MessagesScreenShot",
+        iPadPro11:    "MZPFT.SortedJ317MessagesScreenShot",
```
Comment (janpio) [inline spaceship/lib/spaceship/du/du_client.rb]:
```suggestion
        iPadPro11:    "MZPFT.SortedJ317MessagesScreenShot",
```

File: spaceship/lib/spaceship/du/du_client.rb [hunk 2]
```diff
@@ -121,13 +123,15 @@ def messages_picture_type_map
       # rubocop:enable Layout/ExtraSpacing
       {
         ipad:         "MZPFT.SortedTabletMessagesScreenShot",
-        ipadPro:      "MZPFT.SortedJ99MessagesScreenShot",
         ipad105:      "MZPFT.SortedJ207MessagesScreenShot",
+        ipadPro:      "MZPFT.SortedJ99MessagesScreenShot",
+        iPadPro11:    "MZPFT.SortedJ317MessagesScreenShot", 
+        iPadPro129:   "MZPFT.SortedJ320MessagesScreenShot",
```
Comment (janpio) [inline spaceship/lib/spaceship/du/du_client.rb]:
```suggestion
        iPadPro129:   "MZPFT.SortedJ320MessagesScreenShot",
```

**Comment by max-ott** [comment]
@googlebot Yes

**Comment by max-ott** [comment]
I signed it!

**Review by joshdholtz — COMMENTED** [review]
Thanks for getting these changes going! What exact aspects of _fastlane_ does this all touch? I'm not exactly sure what areas I all need to test for these changes 😇

**Comment by joshdholtz** [comment]
@max-ott @janpio Added some commits here to _snapshot_ and _deliver_ to make this flow complete 💪

**Comment by max-ott** [comment]
@googlebot I signed the CLA.

**Comment by mmouseenvelope12** [comment]
@max-ott @janpio Thanks for adding this! A request, if you would consider it: iPad Pro (3rd Gen) 12.9" is currently one of the four default screenshot sizes in the App Store Connect UI. While it's listed as "Optional," the screenshot formats Apple chooses to highlight here are the ones they are pushing for in new submissions, and thus they are formats Apple wants to see when evaluating games for featuring on the App Store. Would it be possible for deliver to automatically populate both the 3rd gen and 2nd gen categories with our 2048x2732 screenshots? Many thanks!

**Comment by max-ott** [comment]
@mmouseenvelope12 Great idea, but I think it does not really make sense to include this by default. Feel free to make a pull request that adds an option to `deliver` to take the exact same screenshots for 2nd and 3rd gen iPad Pro.

**Comment by janpio** [comment]
(And please create an issue for this idea @mmouseenvelope12 - it will be forgotten and overlooked in here. Thanks!)

**Comment by wdcurry** [comment]
Newbie question: my most basic test of a new bare-boned app with fastlane with only one device (iPhone XR) is snapshot'ing to [828, 1792] and thus failing with deliver due to odd size. Where is this size being drawn from?  I cannot find this value in the downloaded repo.  *Asking here as it seems content-relevant and seems so much easier than the templated requirement, my apologies if this disturbs the continuum ;)

**Comment by max-ott** [comment]
@wdcurry there’s no iPhone XR screenshot category on App Store Connect. You’ll need to take screenshots with / from iPhone XS Max. Those are shared even though they are not the same size. The mentioned dimensions are still correct, just not an accepted size on App Store Connect.

**Comment by wdcurry** [comment]
Thanks Max, I had just wiped them to upload successfully but asked as this was included in an “intro” to snapshots demo I was following. Cheers!

**Pull request merged**